custom/enikshay/reports/repeaters.py

```diff
@@ -233,6 +235,24 @@ def _get_voucher_location_ids(self):
         )
         return possible_location_ids
 
+    def get_messaging_event_detail_link(self, messaging_event_id):
+        return (
+            '<a target="_blank" href="/a/%s/reports/message_event_detail/?id=%s">[%s]</a>' %
+            (self.domain, messaging_event_id, messaging_event_id)
+        )
+
+    def get_messaging_event_links(self, voucher_case_id):
+        event_pks = (
+            MessagingEvent
+            .objects
+            .filter(domain=self.domain, messagingsubevent__case_id=voucher_case_id)
+            .values_list('pk', flat=True)
```

Comment:
Do you think it would be valuable to also pull out the `status` here and display that in the link too, so it is easy to see at a glance what the status of the particular message is? I guess these events don't have readable names (like "Voucher Payment Confirmation"), so this might be more confusing. We can add this later if we think it will be useful.

Reply:
status is actually a little complex to calculate because there are three levels where it can be affected - at the MessagingEvent level, the MessagingSubEvent level, and at the SMS delivery level. so i just linked to the messaging event detail record because that computes the status taking all of those into account

Reply:
ok, thanks